src/main/groovy/org/boozallen/plugins/jte/binding/injectors/ApplicationEnvironmentInjector.groovy

```diff
@@ -22,13 +22,24 @@ import org.boozallen.plugins.jte.binding.*
 import org.jenkinsci.plugins.workflow.cps.CpsScript
 import hudson.Extension 
 import jenkins.model.Jenkins
+import com.cloudbees.groovy.cps.NonCPS
 
 @Extension class ApplicationEnvironmentInjector extends TemplatePrimitiveInjector {
 
+    @NonCPS
     static void doInject(TemplateConfigObject config, CpsScript script){
         Class ApplicationEnvironment = getPrimitiveClass()
+        ArrayList createdEnvs = [] 
         config.getConfig().application_environments.each{ name, appEnvConfig ->
-            script.getBinding().setVariable(name, ApplicationEnvironment.newInstance(name, appEnvConfig))
+            def env = ApplicationEnvironment.newInstance(name, appEnvConfig)
+            createdEnvs << env 
```

Comment:
not a major thing but thinking ...
if( createdEnvs.last ) { env.prev = createdEnvs.last; createEnvs.last.next = env }